cognite_toolkit/_cdf_tk/utils/sql_parser.py

```diff
@@ -45,6 +45,20 @@ def destination_columns(self) -> list[str]:
             self.parse()
         return self._destination_columns
 
+    def is_using_data_set(
+        self, data_set_ids: list[int] | None = None, data_set_external_ids: list[str] | None = None
+    ) -> bool:
+        """Check if the SQL query uses any of the provided data set IDs or external IDs."""
+        for data_set_id in data_set_ids or []:
+            if str(data_set_id) in self.query:
+                return True
+        for data_set_external_id in data_set_external_ids or []:
+            if f'dataset_id("{data_set_external_id}")' in self.query:
+                return True
+            elif f"dataset_id('{data_set_external_id}')" in self.query:
```

Comment:
`elif` condition looks like a duplicate

Reply:
Right, did not notice

Reply:
I could do a one liner, but that would be even less readable.

Reply:
not suggesting this is great, but could be
```
for quote in ['\'', '\"']:
    if f'dataset_id({quote}{data_set_external_id}{quote})' in self.query:
        return True
```
when it comes to readability, I found it hard to spot the diff, as the current code does not highlight it